yellow-paper/docs/decentralisation/governance.md

```diff
@@ -0,0 +1,237 @@
+
+#### Results
+If the vote fails, there is no action needed.
+
+If the vote passes, and a new rollup has been determined to be the next canonical instance, it will become canonical in the amount of days defined within the vote's timelock. It is likely there are defined limitations around this parameter, eg it must be a 3-30 day timelock. This is explained more in the timing section below. At this block height, portals that desire to follow governance should start referencing the new canonical instance to ensure as many bridged assets are backed on the latest version as possible.
+
+:::danger
+Question: what is needed to pass a vote? 
+
+Current thikning is that it should likely be the amount expected to be held in the intial instance of the rollup, e.g. 20% circulating & 25% of that is staked -> 5% of total supply, so locked tokens do not need to participate whatsoever in a happy path upgrade to v1.
+:::
+
+:::danger
+Lasse: note that if the portals follow the governance registry blindly, they assume that new inbox/outbox will always be backwards compatible
+:::
+
+### Timing
+#### Phase 1 - Setup
+After the current canonical rollup, or a sufficient number of tokens are locked in governance, there is a ~3-7 day perparation period where users get their tokens "ready" for voting. ie. withdraw & deposit/lock for the vote, or choose a suitable delegate.
+
+#### Phase 2 - Voting 
+After setup has completed, there is a 7-30 day (TBD) period during which votes can land on the governance contract. In practice, we envision a majority of this voting happening in the current canonical instance and the voting weight of the current canonical instance being sufficient to reach quorum without any additional token delegation. 
+
+#### Phase 3 - Execution Delay (Timelock) 
+
+If a vote passes, there is a timelocked period before it becomes the new canonical rollup. This specific time period must be more than a minimum, eg 3 days, but is defined by the current rollup and in v1 may be controled by both the sequencers in a hapy path, and an emergency security council in a worst case scenario (articulated [below](#Emergency-mode)). In a typical happy path scenario, we suggest this is at least 30 days, and in an emergency, the shortest period possible.
+
+:::info
+It is worth acknowledging that this configurability on upgrade delay windows will likely be flagged on L2 beat as a "medium" centralization risk, due to the ability to quickly upgrade the software (eg a vacation attack). Explicitly this decision could cause us to be labeled a "stage 1" rather than "stage 2" rollup. However, if a vote is reasonably long, then it should be fine as you can argue that the "upgrade period" is the aggregate of all 3 periods.
```

Comment:
```suggestion
It is worth acknowledging that this configurability on upgrade delay windows will likely be flagged on L2 beat as a "medium" centralization risk, due to the ability to quickly upgrade the software (e.g., a vacation attack). Explicitly this decision could cause us to be labeled a "stage 1" rather than "stage 2" rollup. However, if a vote is reasonably long, then it should be fine as you can argue that the "upgrade period" is the aggregate of all 3 periods.
```

A note on this, the delay would essentially be defined as part of the implementation of the instance so an instance could use 30 days always etc, while other have other values, essentially makes it more a instance specific valuation.